test/compilable/ddoc_markdown_headings.d

```diff
@@ -0,0 +1,71 @@
+// PERMUTE_ARGS:
+// REQUIRED_ARGS: -D -Dd${RESULTS_DIR}/compilable -o- -transition=markdown
+// POST_SCRIPT: compilable/extra-files/ddocAny-postscript.sh
+
+/++
+# ATX-Style Headings
+
+# H1
+## H2
+### H3
+#### H4
+##### H5
+###### H6
+
+ ### headings
+  ## with initial
+   # spaces
+
+## heading with *emphasis*
+
+## heading with trailing `#`'s #######
+## heading with trailing literal ##'s
+## heading with another trailing literal#
+## heading with backslash-escaped trailing #\##
+
+## Some empty headers:
+##
+#
+### ###
+
+Setext-Style Headings
+=====================
```

Comment:
Does it work to have fewer `=` than the number of letters?

Reply:
Yes—any number will do per the [CommonMark spec](https://spec.commonmark.org/0.28/#setext-headings). In short, the rules are that the line may only contain whitespace and `=` or `*` characters, and all the `=` or `*` characters must be consecutive. And of course the previous line must contain text.